indico/modules/events/abstracts/controllers/email_templates.py

```diff
@@ -0,0 +1,119 @@
+                self.event_new, self.event_new.abstract_email_templates))
+        return jsonify_template('events/abstracts/management/notification_tpl_form.html', form=form)
+
+
+class RHEditEmailTemplateBase(RHManageAbstractsBase):
+    """Base class for operations that involve editing an e-mail template."""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.email_tpl
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageAbstractsBase._checkParams(self, params)
+        self.email_tpl = AbstractEmailTemplate.get_one(request.view_args['email_tpl_id'])
+
+
+class RHEditEmailTemplateRules(RHEditEmailTemplateBase):
+    """Edit the rules of a notification template."""
+
+    def _process(self):
+        form = EmailTemplateRuleFormEdit(obj=self.email_tpl, event=self.event_new)
+        if form.validate_on_submit():
+            form.populate_obj(self.email_tpl)
+            tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+            return jsonify_data(html=tpl.render_notification_list(
+                self.event_new, self.event_new.abstract_email_templates))
+        return jsonify_template('events/abstracts/management/notification_tpl_form.html',
+                                form=form, is_edit=True)
+
+
+class RHEditEmailTemplateText(RHEditEmailTemplateBase):
+    """Edit the e-mail text of a notification template."""
+
+    def _process(self):
+        form = EmailTemplateTextFormEdit(obj=self.email_tpl, event=self.event_new)
+        if form.validate_on_submit():
+            form.populate_obj(self.email_tpl)
+            tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+            return jsonify_data(html=tpl.render_notification_list(
+                self.event_new, self.event_new.abstract_email_templates))
+        return jsonify_template('events/abstracts/management/notification_tpl_text_form.html',
+                                form=form, is_edit=True)
+
+
+class RHDeleteEmailTemplate(RHEditEmailTemplateBase):
+    """Delete an e-mail template."""
+
+    def _process(self):
+        db.session.delete(self.email_tpl)
+        tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+        return jsonify_data(html=tpl.render_notification_list(self.event_new, self.event_new.abstract_email_templates),
+                            flash=False)
+
+
+class RHPreviewEmailTemplate(RHEditEmailTemplateBase):
+    """Preview an e-mail template."""
+
+    def _process(self):
+        with db.session.no_autoflush:
+            abstract = create_mock_abstract(self.event_new)
+        body = replace_placeholders('abstract-notification-email', self.email_tpl.body,
+                                    abstract=abstract, escape_html=False)
+
+        # we don't want all the mock information we've created above to be saved
+        db.session.rollback()
```

Comment:
`self.commit = False` - not sure if a rollback on the SA level won't screw things up with the crappy zodb/sqlalchemy integration